packages/models/src/store/store.ts

```diff
@@ -0,0 +1,179 @@
+import type { ActorMessage, MessagePayload } from '../actor'
+import type { OutPointString, StoreMessage, StorePath, GetStorageStruct } from './interface'
+import type { JSONStorageOffChain } from './json-storage'
+import type { GetState } from './chain-storage'
+import { ChainStorage } from './chain-storage'
+import { Actor } from '../actor'
+import { JSONStorage } from './json-storage'
+import { NonExistentException, NonStorageInstanceException } from '../exceptions'
+
+export class Store<
+  StorageT extends ChainStorage,
+  StorageStruct extends GetStorageStruct<GetState<StorageT>>,
+  Option = never,
+> extends Actor<StorageStruct, MessagePayload<StoreMessage<StorageStruct>>> {
+  constructor(params?: { states?: Record<OutPointString, StorageStruct>; options?: Option }) {
```

Comment:
I forgot the reason for introducing `outpoint` as the key of a storage struct, have we recorded it in an issue?

Reply:
It's used to index cells' storage data easily.